From 354b475d2845b2a0acf55ac6b19e5e9974bca5a4 Mon Sep 17 00:00:00 2001
From: Drew Willcoxon <adw@mozilla.com>
Date: Sun, 16 Jan 2022 00:36:04 +0000
Subject: [PATCH 19/33] Bug 1750205 - Properly handle a11y when the Firefox
 Suggest online opt-in modal changes panes/sections. r=daisuke, a=RyanVM,
 a=dsmith

Updating `aria-labelledby` and `aria-describedby` seems to work. We aren't using
`aria-live` correctly I think, which is my fault since I suggested it. AFAICT
that should be used on an element whose subtree changes. That's not the case for
these `h1` elements. It is the case for the `body` element, and I tried setting
`aria-live=polite` on it, but it ended up causing everything in the main section
to be read multiple times.

Differential Revision: https://phabricator.services.mozilla.com/D136055
---
 .../components/urlbar/content/quicksuggestOnboarding.html | 5 +++--
 .../components/urlbar/content/quicksuggestOnboarding.js   | 8 +-------
 2 files changed, 4 insertions(+), 9 deletions(-)

diff --git b/browser/components/urlbar/content/quicksuggestOnboarding.html a/browser/components/urlbar/content/quicksuggestOnboarding.html
index 72d82bd3a707..cdcdf2debcc6 100644
--- b/browser/components/urlbar/content/quicksuggestOnboarding.html
+++ a/browser/components/urlbar/content/quicksuggestOnboarding.html
@@ -30,6 +30,7 @@
       <span class="logo" role="presentation"></span>
       <h1 id="introduction-title"
           class="title"
+          aria-live="polite"
           data-l10n-id="firefox-suggest-onboarding-introduction-title-1"></h1>
       <div class="buttonBox">
         <button id="onboardingNext"
@@ -47,14 +48,14 @@
       <span class="logo" role="presentation"></span>
       <h1 id="main-title"
           class="title"
+          aria-live="polite"
           data-l10n-id="firefox-suggest-onboarding-main-title-1"></h1>
       <div class="description-section">
         <h2 id="main-description"
             class="description"
             tabindex="-1"
             data-l10n-id="firefox-suggest-onboarding-main-description-1"></h2>
-        <h3 id="main-privacy-first"
-            class="privacy-first"
+        <h3 class="privacy-first"
             tabindex="-1"
             data-l10n-id="firefox-suggest-onboarding-main-privacy-first"></h2>
       </div>
diff --git b/browser/components/urlbar/content/quicksuggestOnboarding.js a/browser/components/urlbar/content/quicksuggestOnboarding.js
index 252164224233..53829192e1f4 100644
--- b/browser/components/urlbar/content/quicksuggestOnboarding.js
+++ a/browser/components/urlbar/content/quicksuggestOnboarding.js
@@ -145,15 +145,9 @@ document.addEventListener("DOMContentLoaded", async () => {
     window.close();
   });
   addSubmitListener(document.getElementById("onboardingNext"), () => {
-    window.arguments[0].visitedMain = true;
     document.getElementById("introduction-section").classList.add("inactive");
     document.getElementById("main-section").classList.add("active");
-    document.body.setAttribute("aria-labelledby", "main-title");
-    let ariaDescribedBy = "main-description";
-    if (variation?.mainPrivacyFirst) {
-      ariaDescribedBy += " main-privacy-first";
-    }
-    document.body.setAttribute("aria-describedby", ariaDescribedBy);
+    window.arguments[0].visitedMain = true;
   });
   addSubmitListener(document.getElementById("onboardingLearnMore"), () => {
     window.arguments[0].choice = ONBOARDING_CHOICE.LEARN_MORE_2;
-- 
2.25.1

